### PR TITLE
New package: sof-firmware-1.5.1

### LINKED ISSUE #24162: Package request: sof-firmware
Package: `sof-firmware`
Description: Firmware related to audio drivers in linux kernel.
Homepage: https://thesofproject.github.io/latest/introduction/index.html
License (from FAQ):
> Most of the host driver code is dual-licensed BSD or GLPLv2 only (user’s choice). The part of the driver that is GPLv2 only is the Linux integration layer at the top of the driver stack.

I looked shamelessly at [gentoo ebuild](https://gitweb.gentoo.org/repo/gentoo.git/tree/sys-firmware/sof-firmware/sof-firmware-1.5.1.ebuild) and it looks easy to implement.

After updating my void to linux5.7 audio stopped working. After looking to the output of `dmesg` I'm guessing that I'm missing sof-firmware (not sure). Hence package request.

### PR DESCRIPTION
**Pull request by cinerea0**
This commit resolves #24162 by adding the package `sof-firmware`.

### CODE REVIEW

File: srcpkgs/sof-bin/template [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	for f in ${intel_path}/sof/v${version}/*.{ldc,ri}; do
+		vinstall ${f} 0644 /usr/${intel_path}/sof
+	done
+	for f in ${intel_path}/sof/v${version}/intel-signed/*; do
+		vinstall ${f} 0644 /usr/${intel_path}/sof/intel-signed
+	done
+	for f in ${intel_path}/sof/v${version}/public-signed/*; do
+		vinstall ${f} 0644 /usr/${intel_path}/sof/public-signed
+	done
+	ln -s ${DESTDIR}/usr/${intel_path}/sof/sof-bdw-v${version}.ri ${DESTDIR}/usr/${intel_path}/sof/sof-bdw.ri
```
Comment (ericonr) [inline srcpkgs/sof-bin/template:28]: `ln -s sof-bdw-v${version}.ri ${DESTDIR}/usr/${intel_path}/sof/sof-bdw.ri`

Do this for all the commands below, where applicable, and, if possible, use a for loop.

File: srcpkgs/sof-bin/template [hunk 2]
```diff
@@ -0,0 +1,42 @@
+# Template file for 'sof-bin'
+pkgname=sof-bin
+version=1.5.1
+revision=1
+archs="i686 x86_64"
+wrksrc=${pkgname}-stable-v${version}
+make_dirs="/usr/lib/firmware/intel/sof/intel-signed 0755 root root
+ /usr/lib/firmware/intel/sof/public-signed 0755 root root
+ /usr/lib/firmware/intel/sof-tplg 0755 root root"
```
Comment (ericonr) [inline srcpkgs/sof-bin/template:9]: These directories already exist in the tarball. `make_dirs` is for empty directories that are cleaned when generating the tarball, and are instead created at package installation time.

File: srcpkgs/sof-bin/template [hunk 3]
```diff
@@ -0,0 +1,42 @@
+# Template file for 'sof-bin'
+pkgname=sof-bin
+version=1.5.1
+revision=1
+archs="i686 x86_64"
```
Comment (ericonr) [inline srcpkgs/sof-bin/template:5]: Why is this limited? If it's only for x86 stuff, it should at least be `i686* x86_64*` to cover musl.

**Comment by ericonr** [comment]
Regarding the package naming, I do prefer `sof-firmware`, but I would wait for other opinions.

File: srcpkgs/sof-bin/template (hunk 2 above)
Comment (cinerea0) [reply srcpkgs/sof-bin/template]: Ah, I misunderstood what the `make_dirs` variable was for. It's been removed.

File: srcpkgs/sof-bin/template (hunk 1 above)
Comment (cinerea0) [reply srcpkgs/sof-bin/template]: I've updated the template to use this format with a for loop. There are a couple where the architecture in the middle (bdw in your example) doesn't match the symlink, so I had to leave them out of the for loop.

File: srcpkgs/sof-bin/template (hunk 3 above)
Comment (cinerea0) [reply srcpkgs/sof-bin/template]: I honestly forgot musl existed while writing the template. It's been fixed.

**Comment by cinerea0** [comment]
I decided on `sof-bin` because it complies with the normal naming conventions for Void packages, but I would also prefer it to be named `sof-firmware` because it better describes the package's contents.

**Comment by cinerea0** [comment]
Sorry about that closing and re-opening; I forgot to actually commit my changes before pushing.

**Comment by sgn** [comment]
License is `BSD-3-Clause`, no?

**Comment by cinerea0** [comment]
@sgn You're correct, I got confused by the other licenses in the LICENCE.NXP file that refer to files that aren't installed by the package. By the way, what's your opinion on the `sof-bin` vs `sof-firmware` discussion above?

**Comment by sgn** [comment]
On 2020-08-10 09:01:28-0700, cinerea0 <notifications@github.com> wrote:
> By the way, what's your opinion on the `sof-bin` vs `sof-firmware` discussion above?

I prefer `sof-firmware`, but it's not a strong opinion :-p

**Comment by cinerea0** [comment]
I decided to change it to `sof-firmware`. If the maintainers would like me to change it back to `sof-bin` before merging, just let me know.

**Comment by kotoko** [comment]
As compromise it could be `sof-firmware-bin` ;)

**Comment by sgn** [comment]
Ping me in 3 days for a merge if noone complains.

**Comment by Vaelatern** [comment]
Saw your ask on IRC, looks good, so merged. Feel free to ping me in IRC for reviews.

**Pull request merged**